### ISSUE TITLE
Test accessiblity with screen readers

### ISSUE THREAD

**Issue by johannesh2** [vaadin/vaadin-cookie-consent#12]
- Test accessibility with NVDA in Windows
- Test accessibility with JAWS in Windows
- Test accessibility with VoiceOver in Mac OS

**Comment by vlukashov** [comment]
1. On Windows 7 (both with IE 11 and Chrome) the cookie consent dialog is added to the front of list of focusable elements, but it does not get the focus. That is, in order to interact with the 'learn more' / 'dismiss' buttons after the page loads, users have to tab through the entire page and then through the entire browser UI before they reach these controls.
   NOTE: on MacOS this works fine. Pressing 'tab' after the page loads moves focus into the cookie consent dialog.

2. NVDA on Win 7 (both with IE 11 and Chrome)
   2.1. when the page loads: NVDA does not announce the cookie consent popup (neither in Chrome nor in IE11)
   2.2. when the focus is moved to the first focusable element in the popup (the 'learn more' link): NVDA reads the content of the dialog twice

   Chrome:
   > Cookie consent dialog. This website uses cookies to ensure you get the best experience. Learn more about cookies. This website uses cookies to ensure you get the best experience. Button "Learn more about cookies".
       
   IE11:
   > Dialog "This website uses cookies to ensure you get the best experience. Learn more." This website uses cookies to ensure you get the best experience. Button "Learn more about cookies".

3. JAWS on Win 7 (both with IE 11 and Chrome)
   3.1. when the page loads: in Chrome JAWS does not announce the cookie consent popup, but in IE11 that works OK.
   3.2. when the focus is moved to the first focusable element in the popup (the 'learn more' link): JAWS does not read the title nor the content of the dialog. Thus, in Chrome the consent dialog text is never read out loud.

4. VoiceOver on macOS 10.13.6 (with Chrome)
   Reads the contents of the cookie consent dialog on page load [OK].
   Produces sane voice over when moving the focus into the dialog [OK].

5. (bonus) also tested accessibility according to the Section 508 and WCAG 2.0 Level A / AA using [aXe](https://github.com/dequelabs/axe-core). Added an automatic test for that.
Currently has one failure:
```
Element has insufficient color contrast of 3.65 (foreground color: #ffffff, background color: #0090c0, font size: 10.8pt, font weight: bold). Expected contrast ratio of 4.5:1
<a aria-label="dismiss cookie message" role="button" tabindex="0" class="cc-btn cc-dismiss">Got it!</a>
```

Status:
- [x] VoiceOver reads the dialog contents on page load in Chrome on MacOS
- [ ] NVDA reads the dialog contents on page load in IE11 on Windows 7
- [ ] NVDA reads the dialog contents on page load in Chrome on Windows 7
- [x] JAWS reads the dialog contents on page load in IE11 on Windows 7
- [ ] JAWS reads the dialog contents on page load in Chrome on Windows 7
- [x] VoiceOver reads the contents of the dialog once when the dialog gets focus in Chrome on MacOS
- [ ] NVDA reads the contents of the dialog once when the dialog gets focus in IE11 / Chrome on Windows 7
- [ ] JAWS reads the contents of the dialog once when the dialog gets focus in IE11 / Chrome on Windows 7
- [ ] The dialog gets focus when pressing 'Tab' after a page load on Windows
- [x] The dialog gets focus when pressing 'Tab' after a page load on MacOS
- [ ] The dialog passes the Section 508 and WCAG 2.0 Level A / AA audit by [aXe](https://github.com/dequelabs/axe-core)

**Comment by tulioag** [comment]
Using NVDA in Windows 7 with Firefox I am getting:

> Cookie consent dialog This website uses cookies to ensure you best the best experience. Learn more about cookies Clickable learn more about cookies button cookie consent button learn more about cookies button dismiss cookie message

If I use the mouse to hover on the cookie consent dialog, the aria labels are ignored on IE11 and Firefox. They work on Chrome. If I hover with the mouse on the message, on IE11 I get:
> This website uses cookies to ensure you get the best experience. Learn more.

On Chrome and Firefox:
> This website uses cookies to ensure you get the best experience.

**Comment by tulioag** [comment]
On Edge + NVDA using tab we get
> Button learn more about cookies. Cookie consent.

The cookie consent message is not read.

**Comment by johannesh2** [comment]
Created #39 and #40. Closing.
